### DISCUSSION TITLE
Likely a small bug

### DISCUSSION THREAD

**Discussion by yom-ko** [sidrao2006/dynamic_cached_fonts#123] [Q&A]
Hello @sidrao2006 !

I have this exception in my production Sentry logs:

```
Ca: Bad state: Font should already be cached to be loaded
  libapp              0x7f8a0b2ad0 RawDynamicCachedFonts.loadCachedFamily (package:dynamic_cached_fonts/src/raw_dynamic_cached_fonts.dart:253)
  File "<asynchronous suspension>"
  libapp              0x7f8a097060 DynamicCachedFonts.load (package:dynamic_cached_fonts/dynamic_cached_fonts.dart:350)
```

It doesn't look like a serious issue, but it's probably worth your attention. 

![load_issue](https://user-images.githubusercontent.com/2071596/129358716-c3410a4b-152c-4182-8777-ddbc0f77b283.png)

**Comment by sidrao2006** [comment]
Hey @yom-ko,

Thanks for reporting this. Can you please state which version of this package are you using?

Am I right to assume that your usage is similar to [the one mentioned in this example](https://github.com/sidrao2006/dynamic_cached_fonts/blob/main/example/lib/src/demos/multi_font_loading_demo.dart#L18-L27)?

Are you able to reproduce the issue locally or on another device?

Can you please enable verbose logging as mentioned below to better track the problem?

Add `DynamicCachedFonts.toggleVerboseLogging(true)` before the command that you wish to log

Thank you!

**Reply by yom-ko** [reply]
@sidrao2006, sorry for the late reply.

**1.** I'm using `dynamic_cached_fonts: ^0.3.0`
**2.**  Not quite. I use the simple variant, ie.
```
fontItems.forEach((_, fontItem) async {
  DynamicCachedFonts.toggleVerboseLogging(true);
  final DynamicCachedFonts dynamicCachedFont = DynamicCachedFonts(
    fontFamily: fontItem.name,
    url: fontItem.url as String,
  );

  await dynamicCachedFont.load();
});
```
**3.** I also encounter the issue during local debugging, but it doesn't seem to affect the app functionality in any way.
**4.** Yeah, I tried it but its output is not very useful in this case:
```
[DynamicCachedFonts] Font file downloaded
                     
                     Validity: 2021-08-14 04:18:52.422436
                     Download URL: https://<super-secret-path>/Mono-Regular.ttf
[DynamicCachedFonts] Font is not in cache.
                     Loading font now...
```  
<br>
Just in case, here is my environment:

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.3, on Microsoft Windows [Version 10.0.19043.1165], locale ru-RU)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1.0)
[✓] VS Code (version 1.59.0)
[✓] Connected device (3 available)

• No issues found!
```

Hope this helps!

**Reply by sidrao2006** [reply]
Hey @yom-ko, thanks for your reply 

Are you able to see

```
[DynamicCachedFonts] Font has been loaded!
```

before the logs that you provided?
